### PR TITLE
NIFI-13175: Updating dialog layout and sizes to prevent a double scroll bar on smaller screen sizes

### PR DESCRIPTION
**Pull request by mcgilman**
NIFI-13175:
- Updating dialog layout and sizes to prevent a double scroll bar on smaller screen sizes. Now the dialog content is set to 50vh and content scrolls as needed.

### CODE REVIEW

**Comment by rfellows** [comment]
reviewing...

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/system-diagnostics-dialog/system-diagnostics-dialog.component.html [hunk 1]
```diff
@@ -19,12 +19,12 @@
     <h2 mat-dialog-title>System Diagnostics</h2>
     <error-banner></error-banner>
     <div class="system-diagnostics">
-        <mat-dialog-content>
-            <div class="dialog-content">
-                <mat-tab-group>
-                    <mat-tab label="JVM">
-                        <div class="tab-content py-4 h-full w-full">
-                            <div class="inset-0 flex gap-y-4">
+        <div class="dialog-content">
```
Comment (rfellows) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/system-diagnostics-dialog/system-diagnostics-dialog.component.html:22]: this `dialog-content` class is not needed and makes the dialog have double scrollbars.
Screenshot 2024-05-09 at 9 40 19 AM

**Review by rfellows — CHANGES_REQUESTED** [review]
Only found 1 thing needing a tweak...

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/create-connection/create-connection.component.scss [hunk 2]
```diff
@@ -20,16 +20,6 @@
 .create-connection-form {
     @include mat.button-density(-1);
 
-    .mdc-dialog__content {
-        padding: 0 16px;
```
Comment (scottyaslan) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/connection/create-connection/create-connection.component.scss:24]: I see this rule is removed for all the tabbed dialog. Do we need to reapply it in `_app.scss`? If not, then can we remove this rule from the other dialogs? There are only four of them:

Screenshot 2024-05-09 at 9 50 05 AM

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/styles/_app.scss [hunk 3]
```diff
@@ -165,6 +170,10 @@
     .mat-mdc-paginator {
         background-color: unset !important;
     }
+
+    .mdc-dialog__content {
```
Comment (scottyaslan) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/styles/_app.scss:174]: So either we want to move the font-size rule under the `.dialog-tab-content` 

```suggestion
    .dialog-tab-content,
    .dialog-content {
        height: 50vh;
        
        .mdc-dialog__content {
             font-size: 14px;
        }
    }

```

and then we don't need to use `!important` or if we want to force all dialog content text to be `font-size:14px` then we can keep it as but please remove the `font-size: 14px` rule from the other dialog style sheets where we are doing things like:

```
.mdc-dialog__content {
        padding: 0 16px;
        font-size: 14px;

.
.
.
    }
```

It looks like there are only five other such dialogs: 
Screenshot 2024-05-09 at 9 52 22 AM

**Review by scottyaslan — COMMENTED** [review]
Initial pass looks good. Minor comment about cleaning up some styles. Will keep testing.

**Comment by scottyaslan** [comment]
I found one dialog that may need a smaller height:

Screenshot 2024-05-09 at 11 37 54 AM

**Review by rfellows — APPROVED** [review]
Very good changes in here. Thanks, @mcgilman 👍

**Pull request merged**